apm-agent-core/src/main/java/co/elastic/apm/impl/transaction/Span.java

```diff
@@ -59,6 +59,7 @@ public Span(ElasticApmTracer tracer) {
     }
 
     public Span start(Transaction transaction, @Nullable Span parentSpan, long epochMicros, boolean dropped) {
+        this.finished = false;
```

Comment:
I think this better go into a final non-abstract `start` method in AbstractSpan, like you did with `end`, and the rest can be an implementation of an abstract method defined in super (like `doEnd`)